pravega/pravega-tools
# Pravega cluster provisioning model

**Change log description**
First version of a provisioning model for Pravega clusters.

**Purpose of the change**
Fixes #14.

**What the code does**
Based on an empirical analysis, we propose a simple yet complete model to figure out the necessary number of instances of each service based on some user input:
- Availability (failures to tolerate)
- Write workload (min number of Bookies)
- Latency requirements (min number of Segment Stores)
- Metadata operations, streams and clients (min number of Controllers)

Note that while the model aims at being generic, the performance-related values correspond to a specific environment (high-end bare metal cluster). For applying this model to other environments, we may need to perform some benchmarks to get the corresponding performance profile that the model takes as input.

**How to verify it**
Execute the script to verify its correct operation. The validity of the model should be checked via the provisioning of clusters following its guidelines, and in parallel, with the execution of real experiments on these cluster that match the workload requirements specified.
